iosApp/iosApp/Pages/Search/Results/SearchResultsContainer.swift

```diff
@@ -65,40 +72,58 @@ struct SearchResultsContainer: View {
                 resultsState = if results.stops.isEmpty, !showRoutes || results.routes.isEmpty {
                     .empty
                 } else {
-                    .results(results: results, includeRoutes: showRoutes)
+                    .results(
+                        stops: results.stops.compactMap { mapStopIdToResult(id: $0.id) },
+                        routes: results.routes,
+                        includeRoutes: showRoutes
+                    )
                 }
             case nil: resultsState = .error
             case let .error(error):
                 resultsState = .error
                 debugPrint(error)
             }
-        }
+        } catch {}
     }
 
-    func loadVisitHistory() {
-        Task {
-            do {
-                latestVisits = try await visitHistoryUsecase.getLatestVisits()
-                    .compactMap { visit in
-                        switch onEnum(of: visit) {
-                        case let .stopVisit(stopVisit):
-                            // TODO: https://app.asana.com/0/1205425564113216/1208355161933391/f
-                            globalResponse?
-                                .stops[stopVisit.stopId]
-                                .map { stop in
-                                    StopResult(
-                                        id: stop.id,
-                                        rank: 0,
-                                        name: stop.name,
-                                        zone: nil,
-                                        isStation: true,
-                                        routes: []
-                                    )
-                                }
-                        }
+    func loadVisitHistory() async {
+        do {
+            latestVisits = try await visitHistoryUsecase.getLatestVisits()
+                .compactMap { visit in
+                    switch onEnum(of: visit) {
+                    case let .stopVisit(stopVisit):
+                        mapStopIdToResult(id: stopVisit.stopId)
                     }
-            } catch {}
-        }
+                }
+        } catch {}
+    }
+
+    private func mapStopIdToResult(id: String) -> Result? {
+        guard let globalResponse, let stop = globalResponse.stops[id] else { return nil }
+        let isStation = stop.locationType == .station
+        let routePills: [RoutePillSpec] = globalResponse.trips
```

Comment:
nitpick: I don't think efficiency is actually that important here, but it might make more sense to filter through either all routes or all route patterns here, rather than trips, since that's a much smaller set of objects. Then we'd only have to look at each of the route pattern's representative trips, rather than every trip.